llvm/lib/SYCLNativeCPUUtils/PrepareSYCLNativeCPU.cpp

```diff
@@ -403,6 +403,11 @@ PreservedAnalyses PrepareSYCLNativeCPUPass::run(Module &M,
   for (auto &OldF : OldKernels) {
     auto *NewF =
         cloneFunctionAndAddParam(OldF, StatePtrType, CurrentStatePointerTLS);
+    NewF->setLinkage(llvm::GlobalValue::LinkageTypes::LinkOnceODRLinkage);
+    if (O3 && !NewF->hasFnAttribute(Attribute::NoInline)) {
```

Comment:
Does this need to be limited to `-O3` or could we do `if (!NewF->hasOptNone() && !NewF->hasFnAttribute(Attribute::NoInline))` ?

(If we do want it to be limited to `-O3`, I think it'd be slightly better to have `OptLevel` as the field that we store in the pass, and check `OptLevel == OptimizationLevel::O3` here, but I'm thinking we want it at other levels too.)